packages/xchain-thorchain/src/util.ts

```diff
@@ -235,7 +236,31 @@ export const buildUnsignedTx = ({
 
   return new cosmosclient.TxBuilder(cosmosSdk, txBody, authInfo)
 }
+export const getGasExpectedForTx = async ({
+  cosmosSDKClient,
+  txBody,
+  privKey,
+}: {
+  cosmosSDKClient: CosmosSDKClient
+  txBody: proto.cosmos.tx.v1beta1.TxBody
+  privKey: proto.cosmos.crypto.secp256k1.PrivKey
```

Comment:
Remove `privKey` and pass `account: proto.cosmos.auth.v1beta1.IBaseAccount` instead. Or `sequence` and `account_number` instead of `account`. This will save request of `cosmosSDKClient.getAccount`, which might been already done by a caller of `getGasExpectedForTx`  (see `client.transfer`)

Reply:
1. i recall trying to invoke the /simulate endpoint with an unsigned tx and getting some kind of error.
2. the problem is that transferOffline() also calls determineGasLimit() and it doesn't have an account, so i just knowingly made another possibly redundant call to getAccount() inside getGasExpectedForTx() to keep it simple.

we can discuss this one further if you think it's better to try to get rid of it.